velox/docs/configs.rst

```diff
@@ -1133,10 +1133,10 @@ Note: These configurations are experimental and subject to change.
      - Type
      - Default Value
      - Description
-   * - cudf.enabled
-     - bool
-     - true
-     - If true, enable cuDF. By default, it is enabled if compiled with cuDF.
+   * - cudf.disabled
```

Comment:
Why not still use cudf.enabled with default true value? And always detect if gpu is available, if not available, fallback to cpu, and log a warning

Reply:
> Double negatives do hurt my brain

I also find it very confusing.

Also agree that having a `.disabled` parameter makes it feel like something you need to opt-out of.

Finally, if the user does set it to some value, I don't think `auto` should second-guess the user and make it run where it shouldn't.

Reply:
ok, how about I switch back the enabled, and then
* if it's unset, we'll default to auto
* if it's enabled, and we don't have a GPU, we'll fail
* if it's disabled, it's disabled
would you be all OK with that?

@devavret as for opt-in: the user already opts-in because the VELOX_ENABLE_CUDF is off by default, in my opinion. if you're compiling with cuDF *and not setting* enabled to anything explicitly, I feel like auto is a good choice. wdyt?

Reply:
the current code does not detect it for each task/plan. it is detected at startup, once.